ntoskrnl/mm/ARM3/expool.c

```diff
@@ -1465,11 +1465,45 @@ ExpExpandBigPageTable(
     ASSERT(KeGetCurrentIrql() == DISPATCH_LEVEL);
 
     /* Make sure we don't overflow */
-    if (!NT_SUCCESS(RtlULongMult(2,
-                                 OldSize * sizeof(POOL_TRACKER_BIG_PAGES),
-                                 &NewSizeInBytes)))
+    if (!Shrink)
+    {
+        if (!NT_SUCCESS(RtlULongMult(2, OldSize, &NewSize)))
+        {
+            DPRINT1("Overflow expanding big page table. Size=%lu\n", OldSize);
+            KeReleaseSpinLock(&ExpLargePoolTableLock, OldIrql);
+            return FALSE;
+        }
+
+        /* Make sure we don't stupidly waste pages */
+        NewSize = ALIGN_DOWN_BY(NewSize, PAGE_SIZE / sizeof(POOL_TRACKER_BIG_PAGES));
+        ASSERT(NewSize > OldSize);
```

Comment:
This will trigger only if `NewSize` is equal to `OldSize`, aka both `NewSize` and `OldSize` are less than `PAGE_SIZE / sizeof(POOL_TRACKER_BIG_PAGES)`.
Can this happen?

Reply:
No, hence the assertion.